at_secondary/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart

```diff
@@ -32,7 +32,15 @@ abstract class AbstractVerbHandler implements VerbHandler {
   Future<void> process(String command, InboundConnection atConnection) async {
     var response = await processInternal(command, atConnection);
     var handler = responseManager.getResponseHandler(getVerb());
-    handler.process(atConnection, response);
+    try {
+      handler.process(atConnection, response);
+    } on Exception {
+      if (handler.isComplete()) {
```

Comment:
// I think we can hold isComplete() for now. For two reasons:

1. This is request scoped
2. Calling handler.isComplete not in the context of process does not add value

My 2 cents:

try {
      handler.process(atConnection, response);
    } on Exception {
      // Undo needs command as input
      await undo(command);
    }